### PR TITLE
[fix bug 1113270] Connect to mailing list

### PR DESCRIPTION
**Pull request by ckprice**
https://bugzilla.mozilla.org/show_bug.cgi?id=1113270

### CODE REVIEW

**Comment by schalkneethling** [comment]
@ckprice I see nothing obvious immediately but, I bet you it is some Angular craziness where that property is not updated in the template until a resize triggers a 'reflow' so to speak. Some lifecycle thing. We might possible use $compile but, that might be overkill.

The alternative is not to use `ng-show="modalEmail.showFailedBtn"` but instead toggle a class on and of to show and hide the message.

File: src/app/lorax/directives/modal-email.tpl.html [hunk 1]
```diff
@@ -6,16 +6,15 @@
...
         <div class="enter-email__user-input cf">
           <div class="enter-email__email">
             <label for="email" class="is-vishidden">{{::modalEmail.content.label}}</label>
             <input name="email" type="email" id="email" autocorrect="off" autocapitalize="off" spellcheck="false" required aria-required="true" placeholder="{{::modalEmail.content.label}}" ng-model="modalEmail.email" ng-change="modalEmail.onInputChange()">
           </div>
           <div class="enter-email__country">
-            <select>
-              <option>United States</option>
-              <option ng-repeat="country in ::modalEmail.countries">{{country.displayName}}</option>
+            <select name="country" ng-model="modalEmail.country">
```
Comment (ckprice) [inline src/app/lorax/directives/modal-email.tpl.html]: @schalkneethling any idea how I can select an item by default? I think the proper answer is to set the default in the controller here. It should be USA.

Comment (schalkneethling) [inline src/app/lorax/directives/modal-email.tpl.html]: here you go:

`<option ng-repeat="country in ::modalEmail.countries" ng-selected="(country.id=='USA') ? 'selected' : ''" value="{{country.id}}">{{country.displayName}}</option>`

**Comment by schalkneethling** [comment]
@ckprice did you have any success with the wacky message behavior or still facing the same problem?

**Comment by ckprice** [comment]
@schalkneethling all fixes are in! review please, free to merge.

File: src/app/lorax/directives/modal-email.js [hunk 2]
```diff
@@ -92,12 +92,40 @@ define(['angular', 'jquery', 'jquery-customselect'], function (angular, $) {
...
+        var controller = this;
+        var email = controller._$scope.modalEmail.email;
+        var country = controller._$scope.modalEmail.country;
+        var lang = 'en'
+        var fmt = 'H';
+        var privacy = 'on';
+        var params = 'newsletters=' + newsletters + '&source_url=' + source_url + '&email=' + email +
+            '&country=' + country + '&lang=' + lang + '&fmt=' + fmt + '&privacy=' + privacy;
+        controller._$scope.modalEmail.showSubmitBtn = false;
+        var r = new XMLHttpRequest();
```
Comment (schalkneethling) [inline src/app/lorax/directives/modal-email.js]: Just because it is in here already and used in other places, perhaps you can redo this to use jQuery?

**Comment by schalkneethling** [comment]
r+ w/c and we also need a rebase. Thanks!

**Comment by schalkneethling** [comment]
We should also open a bug to improve this dialog in some respects. For example, a progress indicator would be good once one clicks on submit. Also the form elements needs labels that is connected to the elements with an id. Tabbing between form elements also currently does not work.

None of these things was introduced by the work here so, just mentioning it here. Will also file a bug with the above information.

**Comment by ckprice** [comment]
@schalkneethling rebased and squashed r?

**Comment by schalkneethling** [comment]
r+

**Pull request merged**